manifests/plugin/dynflow.pp

```diff
@@ -14,75 +14,32 @@
 #
-# $core_listen::           Address to listen on for the dynflow core service
-#
-# $core_port::             Port to use for the local dynflow core service
-#
 # $ssl_disabled_ciphers::  Disable SSL ciphers. For example: ['NULL-MD5', 'NULL-SHA']
 #
 # $tls_disabled_versions:: Disable TLS versions. Version 1.0 is always disabled. For example: ['1.1']
 #
 # $open_file_limit::       Limit number of open files - Only Red Hat Operating Systems with Software Collections.
-#
-# $external_core::         Forces usage of external/internal Dynflow core
 class foreman_proxy::plugin::dynflow (
   Boolean $enabled = $foreman_proxy::plugin::dynflow::params::enabled,
   Foreman_proxy::ListenOn $listen_on = $foreman_proxy::plugin::dynflow::params::listen_on,
   Optional[Stdlib::Absolutepath] $database_path = $foreman_proxy::plugin::dynflow::params::database_path,
   Boolean $console_auth = $foreman_proxy::plugin::dynflow::params::console_auth,
-  String $core_listen = $foreman_proxy::plugin::dynflow::params::core_listen,
-  Integer[0, 65535] $core_port = $foreman_proxy::plugin::dynflow::params::core_port,
   Optional[Array[String]] $ssl_disabled_ciphers = $foreman_proxy::plugin::dynflow::params::ssl_disabled_ciphers,
   Optional[Array[String]] $tls_disabled_versions = $foreman_proxy::plugin::dynflow::params::tls_disabled_versions,
   Integer[1] $open_file_limit = $foreman_proxy::plugin::dynflow::params::open_file_limit,
-  Boolean $external_core = $foreman_proxy::plugin::dynflow::params::external_core,
 ) inherits foreman_proxy::plugin::dynflow::params {
-  if $foreman_proxy::ssl {
-    $core_url = "https://${facts['networking']['fqdn']}:${core_port}"
-  } else {
-    $core_url = "http://${facts['networking']['fqdn']}:${core_port}"
-  }
-
   foreman_proxy::plugin::module { 'dynflow':
     enabled   => $enabled,
     listen_on => $listen_on,
   }
 
-  if $external_core {
-    $service = 'smart_proxy_dynflow_core'
-
-    file { '/etc/smart_proxy_dynflow_core/settings.yml':
-      ensure  => file,
-      content => template('foreman_proxy/plugin/dynflow_core.yml.erb'),
-      require => Foreman_proxy::Plugin['dynflow_core'],
-      notify  => Service[$service],
-    }
-
-    file { '/etc/smart_proxy_dynflow_core/settings.d':
-      ensure  => link,
-      target  => "${foreman_proxy::config_dir}/settings.d",
-      require => Foreman_proxy::Plugin['dynflow_core'],
-      notify  => Service[$service],
-    }
-  } else {
-    $service = 'foreman-proxy'
-  }
-
-  foreman_proxy::plugin { 'dynflow_core':
-    notify  => Service[$service],
-  }
+  $service = 'foreman-proxy'
 
   systemd::service_limits { "${service}.service":
     limits          => {
       'LimitNOFILE' => $open_file_limit,
     },
     restart_service => false,
-    require         => Foreman_proxy::Plugin['dynflow_core'],
     notify          => Service[$service],
   }
-
-  service { 'smart_proxy_dynflow_core':
-    ensure => $external_core,
-    enable => $external_core,
-  }
```

Comment:
instead of removing the code, we could also ensure → absent for a cleanup, what do you think?

Reply:
The service is owned by the package that should be removed, right? Removing the package is the best way to get rid of the service.

Reply:
At least in the last version of the package I have here:

```
# rpm -ql rubygem-smart_proxy_dynflow_core
/usr/share/gems/gems/smart_proxy_dynflow_core-0.4.1
/usr/share/gems/gems/smart_proxy_dynflow_core-0.4.1/Gemfile.in
/usr/share/gems/gems/smart_proxy_dynflow_core-0.4.1/LICENSE
/usr/share/gems/gems/smart_proxy_dynflow_core-0.4.1/lib
/usr/share/gems/gems/smart_proxy_dynflow_core-0.4.1/lib/smart_proxy_dynflow_core
/usr/share/gems/gems/smart_proxy_dynflow_core-0.4.1/lib/smart_proxy_dynflow_core.rb
/usr/share/gems/gems/smart_proxy_dynflow_core-0.4.1/lib/smart_proxy_dynflow_core/task_launcher_registry.rb
/usr/share/gems/gems/smart_proxy_dynflow_core-0.4.1/lib/smart_proxy_dynflow_core/version.rb
/usr/share/gems/gems/smart_proxy_dynflow_core-0.4.1/smart_proxy_dynflow_core.gemspec
/usr/share/gems/specifications/smart_proxy_dynflow_core-0.4.1.gemspec
```

Reply:
0.4.1 was already an compatibility-only shim mapping constants to dfifferent names. From a version we shipped in 2.4:
```
# rpm -ql ./tfm-rubygem-smart_proxy_dynflow_core-0.3.3-1.fm2_4.el7.noarch.rpm                                                                                                                                                                                    
/etc/logrotate.d/tfm-rubygem-smart_proxy_dynflow_core
/etc/smart_proxy_dynflow_core/settings.yml
/opt/theforeman/tfm/root/usr/share/gems/gems/smart_proxy_dynflow_core-0.3.3
/opt/theforeman/tfm/root/usr/share/gems/gems/smart_proxy_dynflow_core-0.3.3/Gemfile.in
/opt/theforeman/tfm/root/usr/share/gems/gems/smart_proxy_dynflow_core-0.3.3/LICENSE
/opt/theforeman/tfm/root/usr/share/gems/gems/smart_proxy_dynflow_core-0.3.3/bin
/opt/theforeman/tfm/root/usr/share/gems/gems/smart_proxy_dynflow_core-0.3.3/bin/smart_proxy_dynflow_core
/opt/theforeman/tfm/root/usr/share/gems/gems/smart_proxy_dynflow_core-0.3.3/config
/opt/theforeman/tfm/root/usr/share/gems/gems/smart_proxy_dynflow_core-0.3.3/config/settings.yml.example
/opt/theforeman/tfm/root/usr/share/gems/gems/smart_proxy_dynflow_core-0.3.3/lib
/opt/theforeman/tfm/root/usr/share/gems/gems/smart_proxy_dynflow_core-0.3.3/lib/smart_proxy_dynflow_core
/opt/theforeman/tfm/root/usr/share/gems/gems/smart_proxy_dynflow_core-0.3.3/lib/smart_proxy_dynflow_core.rb
/opt/theforeman/tfm/root/usr/share/gems/gems/smart_proxy_dynflow_core-0.3.3/lib/smart_proxy_dynflow_core/api.rb
/opt/theforeman/tfm/root/usr/share/gems/gems/smart_proxy_dynflow_core-0.3.3/lib/smart_proxy_dynflow_core/bundler_helper.rb
/opt/theforeman/tfm/root/usr/share/gems/gems/smart_proxy_dynflow_core-0.3.3/lib/smart_proxy_dynflow_core/callback.rb
/opt/theforeman/tfm/root/usr/share/gems/gems/smart_proxy_dynflow_core-0.3.3/lib/smart_proxy_dynflow_core/core.rb
/opt/theforeman/tfm/root/usr/share/gems/gems/smart_proxy_dynflow_core-0.3.3/lib/smart_proxy_dynflow_core/helpers.rb
/opt/theforeman/tfm/root/usr/share/gems/gems/smart_proxy_dynflow_core-0.3.3/lib/smart_proxy_dynflow_core/launcher.rb
/opt/theforeman/tfm/root/usr/share/gems/gems/smart_proxy_dynflow_core-0.3.3/lib/smart_proxy_dynflow_core/log.rb
/opt/theforeman/tfm/root/usr/share/gems/gems/smart_proxy_dynflow_core-0.3.3/lib/smart_proxy_dynflow_core/logger_middleware.rb
/opt/theforeman/tfm/root/usr/share/gems/gems/smart_proxy_dynflow_core-0.3.3/lib/smart_proxy_dynflow_core/middleware
/opt/theforeman/tfm/root/usr/share/gems/gems/smart_proxy_dynflow_core-0.3.3/lib/smart_proxy_dynflow_core/middleware/keep_current_request_id.rb
/opt/theforeman/tfm/root/usr/share/gems/gems/smart_proxy_dynflow_core-0.3.3/lib/smart_proxy_dynflow_core/request_id_middleware.rb
/opt/theforeman/tfm/root/usr/share/gems/gems/smart_proxy_dynflow_core-0.3.3/lib/smart_proxy_dynflow_core/settings.rb
/opt/theforeman/tfm/root/usr/share/gems/gems/smart_proxy_dynflow_core-0.3.3/lib/smart_proxy_dynflow_core/task_launcher_registry.rb
/opt/theforeman/tfm/root/usr/share/gems/gems/smart_proxy_dynflow_core-0.3.3/lib/smart_proxy_dynflow_core/testing.rb
/opt/theforeman/tfm/root/usr/share/gems/gems/smart_proxy_dynflow_core-0.3.3/lib/smart_proxy_dynflow_core/version.rb
/opt/theforeman/tfm/root/usr/share/gems/gems/smart_proxy_dynflow_core-0.3.3/smart_proxy_dynflow_core.gemspec
/opt/theforeman/tfm/root/usr/share/gems/specifications/smart_proxy_dynflow_core-0.3.3.gemspec
/opt/theforeman/tfm/root/usr/share/smart_proxy_dynflow_core/bundler.d
/usr/bin/smart_proxy_dynflow_core
/usr/lib/systemd/system/smart_proxy_dynflow_core.service
```

Reply:
but it did own that file before, so it's probably cleaned up on removal/upgrade, you're right

https://github.com/theforeman/foreman-packaging/pull/6753/files